"panic: invalid memory address or nil pointer dereference" with uniq

## tl;dr

```
$ super -version
Version: a3f7d1558

$ echo '1 2' | super -s -c "collect([(unnest [1,2] | uniq)])" -
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd0a247e]

goroutine 65 [running]:
github.com/brimdata/super/runtime/sam/op/uniq.(*Op).appendUniq(0xc00039b490, {0x0, 0x0, 0x0}, 0xc0004ae000)
	/Users/phil/work/super/runtime/sam/op/uniq/uniq.go:49 +0x1fe
github.com/brimdata/super/runtime/sam/op/uniq.(*Op).Pull(0xc00039b490, 0x0)
	/Users/phil/work/super/runtime/sam/op/uniq/uniq.go:83 +0x237
github.com/brimdata/super/runtime/sam/op/aggregate.(*Op).run(0xc000451e00)
	/Users/phil/work/super/runtime/sam/op/aggregate/aggregate.go:195 +0xa2
created by github.com/brimdata/super/runtime/sam/op/aggregate.(*Op).Pull.func1 in goroutine 38
	/Users/phil/work/super/runtime/sam/op/aggregate/aggregate.go:163 +0x66
```

## Details

Repro is with super commit a3f7d15.

This issue was reported in a [community Slack thread](https://super-db.slack.com/archives/CTSMAK6G7/p1764563619498789). The user explained this is a simplification of more complex query.

--- a/runtime/sam/op/uniq/uniq.go
+++ b/runtime/sam/op/uniq/uniq.go
@@ -74,6 +74,7 @@ func (o *Op) Pull(done bool) (sbuf.Batch, error) {
 			}
 			o.eos = true
 			t := o.wrap(o.last)
+			o.count = 0
 			o.last = nil
 			return sbuf.NewArray([]super.Value{t}), nil
 		}

--- a/runtime/ztests/op/uniq-subquery.yaml
+++ b/runtime/ztests/op/uniq-subquery.yaml
@@ -1,0 +1,11 @@
+spq: values [(values this,this,this+1,this+1 | uniq)]
+
+vector: true
+
+input: |
+  1
+  2
+
+output: |
+  [1,2]
+  [2,3]

